app/assets/javascripts/discourse/app/components/sidebar/user/custom-sections.js

```diff
@@ -3,28 +3,54 @@ import { inject as service } from "@ember/service";
 import { ajax } from "discourse/lib/ajax";
 import { bind } from "discourse-common/utils/decorators";
 import Section from "discourse/components/sidebar/user/section";
+import CommunitySection from "discourse/components/sidebar/common/community-section";
 
 export default class SidebarUserCustomSections extends Component {
   @service currentUser;
   @service router;
   @service messageBus;
+  @service appEvents;
+  @service topicTrackingState;
+  @service siteSettings;
 
   constructor() {
     super(...arguments);
     this.messageBus.subscribe("/refresh-sidebar-sections", this._refresh);
+    this.callbackIds = [];
   }
 
   willDestroy() {
     this.messageBus.unsubscribe("/refresh-sidebar-sections");
+    this.callbackIds.forEach((id) => {
+      this.topicTrackingState.offStateChange(id);
```

Comment:
I think this needs to be handled by the `CommunitySection` because not every section will be trying to unsubscribe from `topicTrackingState`

Reply:
Actually I see that we're calling `section.teardown` after, I think the section's teardown function can handle the unsubscription from topic tracking state?